tests/tests/bitasset_tests.cpp

```diff
@@ -1407,4 +1407,138 @@ BOOST_AUTO_TEST_CASE(hf_890_test_hf1270)
+         BOOST_CHECK(smartbit_bitasset_data.options.short_backing_asset == jillcoin.id);
+
+         // Fund balances of the actors
+         issue_uia(alice, jillcoin.amount(5000 * std::pow(10, jillcoin.precision)));
+         BOOST_REQUIRE_EQUAL(get_balance(alice, jillcoin), 5000 * std::pow(10, jillcoin.precision));
+         BOOST_REQUIRE_EQUAL(get_balance(alice, smartbit), 0);
+
+
+         /**
+          * Claim any amount of collateral asset fees.  This should fail because claiming such fees are prohibited
+          * before the HARDFORK_CORE_BSIP_87_74_COLLATFEE_TIME.
+          */
+         trx.clear();
+         asset_claim_fees_operation claim_op;
+         claim_op.issuer = smartissuer.id;
+         claim_op.extensions.value.claim_from_asset_id = smartbit.id;
+         claim_op.amount_to_claim = jillcoin.amount(5 * std::pow(10, jillcoin.precision));
+         trx.operations.push_back(claim_op);
+         set_expiration(db, trx);
+         sign(trx, smartissuer_private_key);
+         GRAPHENE_REQUIRE_THROW(PUSH_TX(db, trx), fc::exception);
+
+
+         /**
+          * Advance to when the collateral fee container is activated
+          */
+         generate_blocks(HARDFORK_CORE_BSIP_87_74_COLLATFEE_TIME);
+         generate_block(); trx.clear(); set_expiration(db, trx);
+
+
+         /**
+          * Cause some collateral of JCOIN to be accumulated as collateral fee within the SMARTBIT asset type
+          */
+         // HACK: Before BSIP74 or BSIP87 are introduced, it is not formally possible to accumulate collateral fees.
+         // Therefore, the accumulation for this test will be informally induced by direct manipulation of the database.
+         // More formal tests will be provided with the PR for either BSIP74 or BSIP87.
+         // IMPORTANT: The use of this hack requires that no additional blocks are subsequently generated!
+         asset accumulation_amount = jillcoin.amount(40 * std::pow(10, jillcoin.precision)); // JCOIN
+         db.adjust_balance(alice.id, -accumulation_amount); // Deduct 40 JCOIN from alice as a "collateral fee"
+         smartbit.accumulate_fee(db, accumulation_amount); // Add 40 JCOIN from alice as a "collateral fee"
+
+
+         /**
+          * Attempt to change the backing asset.  This should fail because there are unclaimed collateral fees.
+          */
+         trx.clear();
+         asset_update_bitasset_operation change_backing_asset_op;
+         change_backing_asset_op.asset_to_update = smartbit.id;
+         change_backing_asset_op.issuer = smartissuer.id;
+         change_backing_asset_op.new_options.short_backing_asset = izzycoin.id;
+         trx.operations.push_back(change_backing_asset_op);
+         sign(trx, smartissuer_private_key);
+         REQUIRE_EXCEPTION_WITH_TEXT(PUSH_TX(db, trx), "Must claim collateral-denominated fees");
+
+
+         /**
+          * Attempt to claim a negative amount of the collateral asset fees.
```

Comment:
Attempt to claim `0` too? Perhaps there is already a test case for it elsewhere, I didn't check.